packages/kyt-starter-universal/starter-src/src/routes/index.js

```diff
@@ -1,43 +1,58 @@
-import React from 'react';
-import Route from 'react-router/lib/Route';
-import IndexRoute from 'react-router/lib/IndexRoute';
-import App from '../components/App';
-
-// Webpack 2 supports ES2015 `import()` by auto-
-// chunking assets. Check out the following for more:
-// https://webpack.js.org/guides/migrating/#code-splitting-with-es2015
-
-const importHome = (nextState, cb) => {
-  import(/* webpackChunkName: "home" */ '../components/Home')
-    .then(module => cb(null, module.default))
-    .catch(e => {
-      throw e;
-    });
-};
+import React, { PropTypes } from 'react';
+import { Route, Switch, Redirect } from 'react-router-dom';
+import Home from '../components/Home';
+import Tools from '../components/Tools';
 
-const importTools = (nextState, cb) => {
-  import(/* webpackChunkName: "tools" */ '../components/Tools')
-    .then(module => cb(null, module.default))
-    .catch(e => {
-      throw e;
-    });
-};
+// with guidance from https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/guides/server-rendering.md
+const Status = ({ code, children }) => (
+  <Route
+    render={({ staticContext }) => {
+      // there is no `staticContext` on the client, so
+      // we need to guard against that here
+      if (staticContext) {
+        staticContext.status = code;
+      }
+
+      return children;
+    }}
+  />
+);
+
+const RedirectWithStatus = ({ from, to, status }) => (
+  <Status code={status}>
+    <Redirect from={from} to={to} />
+  </Status>
+);
 
-// We use `getComponent` to dynamically load routes.
-// https://github.com/reactjs/react-router/blob/master/docs/guides/DynamicRouting.md
-const routes = (
-  <Route path="/" component={App}>
-    <IndexRoute getComponent={importHome} />
-    <Route path="tools" getComponent={importTools} />
-  </Route>
+const NotFound = () => (
+  <Status code={404}>
+    <div>
+      <h1>Sorry, can’t find that.</h1>
+    </div>
+  </Status>
 );
 
-// Unfortunately, HMR breaks when we dynamically resolve
-// routes so we need to require them here as a workaround.
-// https://github.com/gaearon/react-hot-loader/issues/288
-if (module.hot) {
-  require('../components/Home'); // eslint-disable-line global-require
-  require('../components/Tools'); // eslint-disable-line global-require
+// Note: A redirect can be added in this form
+// <RedirectWithStatus status={301} from="/old" to="/new" />
+function Routes() {
+  return (
+    <Switch>
+      <Route exact path="/" component={Home} />
+      <Route path="/tools" component={Tools} />
+      <Route component={NotFound} />
+    </Switch>
+  );
 }
 
-export default routes;
+Status.propTypes = {
+  code: PropTypes.number.isRequired,
+  children: PropTypes.node.isRequired,
+};
+
+RedirectWithStatus.propTypes = {
+  from: PropTypes.string.isRequired,
+  to: PropTypes.string.isRequired,
+  status: PropTypes.number.isRequired,
+};
+
+export default Routes();
```

Comment:
does this need to be a singleton?

Reply:
Good catch @staylor, thx. I've updated this.